### ISSUE TITLE
Native crash: `invalid chunk state when deallocating address`

### ISSUE THREAD

**Issue by ByteHamster** [mr-kenhoff/qalculate-android#3]
The app sometimes crashes when I type a lot (with PR #2, every key press re-calculates the result). I don't think it is a race condition because it sometimes happens even when I type slowly. Unfortunately, I do not know enough about the internals of qalculate to know what is going on there. See stack trace below.

```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
LineageOS Version: '18.1-20211018-NIGHTLY-oneplus3'
Build fingerprint: 'OnePlus/OnePlus3/OnePlus3T:8.0.0/OPR1.170623.032/02281230:user/release-keys'
Revision: '0'
ABI: 'arm64'
Timestamp: 2022-03-27 16:28:14+0200
pid: 26090, tid: 26111, name: FinalizerDaemon  >>> com.mrkenhoff.qalculate <<<
uid: 10495
signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'Scudo ERROR: invalid chunk state when deallocating address 0x0071b0311e50

    x0  0000000000000000  x1  00000000000065ff  x2  0000000000000006  x3  000000706d882800
    x4  0000808080808080  x5  0000808080808080  x6  0000808080808080  x7  0000000000000018
    x8  00000000000000f0  x9  e546767144193d9f  x10 ffffff80fffffbdf  x11 ffffffc0fffffbdf
    x12 0000000000000010  x13 00002bb2ecd0fa0a  x14 000d8b21daa5c6dd  x15 000000001c10e206
    x16 00000073141cec78  x17 00000073141b0b90  x18 000000701c8ac000  x19 00000000000000ac
    x20 00000000000065ea  x21 00000000000000b2  x22 00000000000065ff  x23 00000000ffffffff
    x24 000000720026fcd0  x25 000000720026fe40  x26 0000000000000021  x27 0000000000000002
    x28 000000706d882a40  x29 000000706d882880
    lr  00000073141624e0  sp  000000706d8827e0  pc  000000731416250c  pst 0000000000000000
backtrace:
      #00 pc 000000000004e50c  /apex/com.android.runtime/lib64/bionic/libc.so (abort+176) (BuildId: cc1796f800d4bb1887f9ddeae855da7b)
      #01 pc 00000000000428c8  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::die()+8) (BuildId: cc1796f800d4bb1887f9ddeae855da7b)
      #02 pc 0000000000042f74  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::ScopedErrorReport::~ScopedErrorReport()+32) (BuildId: cc1796f800d4bb1887f9ddeae855da7b)
      #03 pc 00000000000431a8  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::reportInvalidChunkState(scudo::AllocatorAction, void*)+76) (BuildId: cc1796f800d4bb1887f9ddeae855da7b)
      #04 pc 0000000000044668  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::Allocator<scudo::AndroidConfig, &(scudo_malloc_postinit)>::deallocate(void*, scudo::Chunk::Origin, unsigned long, unsigned long)+340) (BuildId: cc1796f800d4bb1887f9ddeae855da7b)
      #05 pc 0000000000116084  /data/app/~~F9KFjCUuPh0FwEGCb9dLww==/com.mrkenhoff.qalculate-3cvm13NN_7gOKCEJ7kM1LA==/lib/arm64/libqalculate.so (Calculator::~Calculator()+156)
      #06 pc 0000000000116cb0  /data/app/~~F9KFjCUuPh0FwEGCb9dLww==/com.mrkenhoff.qalculate-3cvm13NN_7gOKCEJ7kM1LA==/lib/arm64/libqalculate.so (Calculator::~Calculator()+16)
      #07 pc 0000000000047f68  /data/app/~~F9KFjCUuPh0FwEGCb9dLww==/com.mrkenhoff.qalculate-3cvm13NN_7gOKCEJ7kM1LA==/lib/arm64/liblibqalculate_swig.so (Java_com_mrkenhoff_libqalculate_libqalculateJNI_delete_1Calculator+76) (BuildId: 2218d54da22db12bf231cec6b50458439db6a7ef)
      #08 pc 000000000013ced4  /apex/com.android.art/lib64/libart.so (art_quick_generic_jni_trampoline+148) (BuildId: b20c6792f7bf2deaba93c26c28652892)
      #09 pc 00000000001337e8  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+568) (BuildId: b20c6792f7bf2deaba93c26c28652892)
      #10 pc 00000000001a7894  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+228) (BuildId: b20c6792f7bf2deaba93c26c28652892)
```

**Comment by ByteHamster** [comment]
Given that it crashes somewhere in the destructor, I guess your idea to make the calculator a Singleton will fix this. We'll see :)

**Comment by mr-kenhoff** [comment]
Fixed in 5b4059efee4ab7f87f27862053e38988b204e957 by not creating and destroying the calculator object every time.
